File: AVc/app/src/main/java/com/hfad/avc/ui/database/TemplateDao.java
package com.hfad.avc.ui.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
@Dao
public interface TemplateDao {
    @Query("SELECT * FROM template")
    List<Template> getAll();

    @Query("SELECT * FROM template WHERE id = :id")
    Template getById(String id);

    @Query("SELECT COUNT(*) FROM template")
    String getSize();

    @Insert
    void insert(Template template);

    @Update
    void update(Template template);

    @Delete
    void delete(Template template);
}

File: app/src/main/java/com/hfad/avc/data/database/TemplateDao.kt
package com.hfad.avc.data.database

import androidx.room.*
import com.hfad.avc.data.model.Contact
import com.hfad.avc.data.model.Template

@Dao
interface TemplateDao {
   @get:Query("SELECT * FROM template")
   val all: List<Template>

   @get:Query("SELECT * FROM template WHERE (favorite != 0)")
   val favorite: List<Template>

   @Query("SELECT * FROM template WHERE id = :id")
   fun getById(id: String): Template?

   @get:Query("SELECT COUNT(*) FROM template")
   val size: String

   @Insert
   fun insert(template: Template)

   @Update
   fun update(template: Template)

   @Upsert
   fun upsert(template: Template)

   @Delete
   fun delete(template: Template)
}